### PR TITLE
HOSTEDCP-2040: Add README.md for ControlPlaneComponent

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: support/controlplane-component/README.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+```
+
+Note that the container image `image-key` will be replaced with the image value corresponding to that key from the release payload if that key exist, and kept as is otherwise.
+
+!!! note 
+
+    If your workload is a StatefulSet, create a filed named `statefulset.yaml` containing the StatefulSet's manifest instead!
+
+
+### Creating other resources manifests
```
Comment (csrwng) [inline support/controlplane-component/README.md:42]: s/resources/resource/

File: support/controlplane-component/README.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+Place any other required resources under the same directory, such as configMaps, secrets, roles, etc. All resources manifests will be automatically be picked up and deployed.
```
Comment (csrwng) [inline support/controlplane-component/README.md:44]: s/resources/resource/

File: support/controlplane-component/README.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+```go hl_lines="6 7 8 9"
+// control-plane-operator/controllers/hostedcontrolplane/v2/mycomponent/component.go
+
+func NewComponent() component.ControlPlaneComponent {
+	return component.NewDeploymentComponent(ComponentName, &MyComponent{}).
+        WithAdaptFunction(adaptDeployment).
+        WithManifestAdapter(
+			"my-config.yaml",
+			component.WithAdaptFunction(adaptMyConfig),
```
Comment (csrwng) [inline support/controlplane-component/README.md:187]: is it necessary to cast to the WithAdaptFunction type?

**Review by csrwng — COMMENTED** [review]
some nits, otherwise lgtm

File: support/controlplane-component/README.md (hunk 3 above)
Comment (muraee) [reply support/controlplane-component/README.md:187]: WithAdaptFunction dosn't expect any specific type, it accepts any function with the following signature `[T client.Object] func(cpContext ControlPlaneContext, resource T) error` 
It is the responsibility of the components' developer to pass a function with the correct type.

**Pull request merged**